stdlib.sh

```diff
@@ -1068,6 +1068,33 @@ layout_pyenv() {
   [[ -n "$PYENV_VERSION" ]] && export PYENV_VERSION
 }
 
+# Usage: layout uv
+#
+# Enables the uv project layout in the current directory, and syncs
+# the dependencies in the project.
+#
+# This relies on the `uv` command being available in the PATH, and performs a
+# sync on cd because uv is fast enough it's not impactful. It relies on uv's
+# configuration file and environment variables, rather than arguments.
+#
+layout_uv() {
+  # Watch the uv configuration file for changes
+  watch_file .python-version
+  watch_file pyproject.toml
+  watch_file uv.lock
+
+  uv sync || return 1
```

Comment:
(I was tinkering with `uv sync --offline`, which could address some concerns about "the network not being as fast as uv". Probably not a good idea)

I think that we could have something more refined here. Here's a scenario:

- `.envrc` like so
```shell
strict_env
export UV_PROJECT_ENVIRONMENT=.fancy-venv
layout uv
```
- some network errors make `uv` fail
- the virtual environment is still created, but it's not activated and the `UV_PROJECT_ENVIRONMENT` is not set (that means that running `uv sync` in the resulting shell would use `.venv` instead of `.fancy-venv`).

I think that this should be a little more forgiving.

What about starting the script (before the `watch_file`) by checking that we could actually do something:

```shell
  if ! has uv; then
    echo "uv is required. Follow this steps to install it: https://docs.astral.sh/uv/getting-started/installation/"
    return 1
  fi
```

And then just assume that the error message is clear enough, but still trying to activate the venv and keeping all the other variables:

```suggestion
  uv sync || true
```

Reply:
Hm. I think I like your idea. I'm going to tweak it a tiny bit, though, to also avoid re-locking (so direnv doesn't ever change project files).